File: EVSTAR/packages/AjaxUploader.4.0.2.7/content/aspnet/advanced_vb/App_Code/FileItem.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls

Public Class FileItem
    Public Sub New(ByVal dir As String, ByVal item As CuteWebUI.AttachmentItem)
        _dir = dir
        _guid = item.FileGuid
        _name = item.FileName
        _size = item.FileSize
        _tempname = item.GetTempFilePath()
    End Sub

    Public Sub New(ByVal dir As String)
        _dir = dir
    End Sub

    Private _dir As String
    Private _guid As Guid
    Private _name As String
    Private _size As Integer
    Private _tempname As String

    Public ReadOnly Property Directory() As String
        Get
            Return _dir
        End Get
    End Property

    Public Property FileGuid() As Guid
        Get
            Return _guid
        End Get
        Set(ByVal value As Guid)
            _guid = value
        End Set
    End Property
    Public Property FileName() As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            _name = value
        End Set
    End Property
    Public Property FileSize() As Integer
        Get
            Return _size
        End Get
        Set(ByVal value As Integer)
            _size = value
        End Set
    End Property
    Public Property TempFileName() As String
        Get
            Return _tempname
        End Get
        Set(ByVal value As String)
            _tempname = value
            _temppath = Nothing
        End Set
    End Property

    Private _temppath As String
    Public ReadOnly Property TempFilePath() As String
        Get
            If _temppath Is Nothing Then
                _temppath = System.IO.Path.Combine(_dir, _tempname)
            End If
            Return _temppath
        End Get
    End Property
End Class

File: EVSTAR/packages/AjaxUploader.4.0.2.7/content/aspnet/advanced_cs/App_Code/FileItem.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

public class FileItem
{
    public FileItem(string dir, CuteWebUI.AttachmentItem item)
	{
		_dir=dir;
		_guid = item.FileGuid;
		_name = item.FileName;
		_size = item.FileSize;
		_tempname = item.GetTempFilePath();
	}

	public FileItem(string dir)
	{
		_dir=dir;
	}

	string _dir;
	Guid _guid;
	string _name;
	int _size;
	string _tempname;

	public string Directory
	{
		get
		{
			return _dir;
		}
	}

	public Guid FileGuid
	{
		get
		{
			return _guid;
		}
		set
		{
			_guid = value;
		}
	}
	public string FileName
	{
		get
		{
			return _name;
		}
		set
		{
			_name = value;
		}
	}
	public int FileSize
	{
		get
		{
			return _size;
		}
		set
		{
			_size = value;
		}
	}
	public string TempFileName
	{
		get
		{
			return _tempname;
		}
		set
		{
			_tempname = value;
			_temppath = null;
		}
	}

	string _temppath;
	public string TempFilePath
	{
		get
		{
			if(_temppath==null)
				_temppath=System.IO.Path.Combine(_dir, _tempname);
			return _temppath;
		}
	}
}
